### DISCUSSION TITLE
Alpine-ESPHome LXC compile errors

### DISCUSSION THREAD

**Discussion by ntguest** [tteck/Proxmox#1256] [Help Needed]
**Describe the issue**
Can't compile any projects both esp32 and esp8266

**Screenshots**
![image](https://user-images.githubusercontent.com/69485846/227750452-f0aa3bcb-34f4-449e-9213-1eb25ce274f7.png)

INFO Reading configuration /root/esphome/config/bleproxy02.yaml...
INFO Updating https://github.com/ntguest/esphome@main
INFO Generating C++ source...
INFO Compiling app...
Processing bleproxy02 (board: esp32dev; framework: arduino; platform: platformio/espressif32 @ 5.2.0)
--------------------------------------------------------------------------------
Library Manager: Installing esphome/AsyncTCP-esphome @ 1.2.2
INFO Installing esphome/AsyncTCP-esphome @ 1.2.2
Unpacking  [####################################]  100%
Library Manager: AsyncTCP-esphome@1.2.2 has been installed!
INFO AsyncTCP-esphome@1.2.2 has been installed!
Library Manager: Installing esphome/ESPAsyncWebServer-esphome @ 2.1.0
INFO Installing esphome/ESPAsyncWebServer-esphome @ 2.1.0
Unpacking  [####################################]  100%
Library Manager: ESPAsyncWebServer-esphome@2.1.0 has been installed!
INFO ESPAsyncWebServer-esphome@2.1.0 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/noise-c @ 0.1.4
INFO Installing esphome/noise-c @ 0.1.4
Unpacking  [####################################]  100%
Library Manager: noise-c@0.1.4 has been installed!
INFO noise-c@0.1.4 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/libsodium @ 1.10018.1
INFO Installing esphome/libsodium @ 1.10018.1
Unpacking  [####################################]  100%          
Library Manager: libsodium@1.10018.1 has been installed!
INFO libsodium@1.10018.1 has been installed!
Library Manager: Installing bblanchon/ArduinoJson @ 6.18.5
INFO Installing bblanchon/ArduinoJson @ 6.18.5
Unpacking  [####################################]  100%
Library Manager: ArduinoJson@6.18.5 has been installed!
INFO ArduinoJson@6.18.5 has been installed!
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch3
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
Dependency Graph
|-- AsyncTCP-esphome @ 1.2.2
|-- WiFi @ 2.0.0
|-- FS @ 2.0.0
|-- Update @ 2.0.0
|-- ESPAsyncWebServer-esphome @ 2.1.0
|   |-- AsyncTCP-esphome @ 1.2.2
|-- DNSServer @ 2.0.0
|-- ESPmDNS @ 2.0.0
|-- noise-c @ 0.1.4
|   |-- libsodium @ 1.10018.1
|-- ArduinoJson @ 6.18.5
Compiling .pioenvs/bleproxy02/src/esphome/components/api/api_connection.cpp.o
sh: xtensa-esp32-elf-g++: not found
*** [.pioenvs/bleproxy02/src/esphome/components/api/api_connection.cpp.o] Error 127
========================= [FAILED] Took 15.82 seconds =========================

**Comment by tteck** [comment]
I am uncertain whether this is related to the script, as the ESPHome dependencies for Alpine and Debian appear to be close to identical.

**Comment by ntguest** [comment]
it looks like esphome doesn't works on alpine.
so, this case is useless(

**Comment by tteck** [comment]
What system are you using to run ESPHome? After doing a quick search on Google, I came across multiple posts reporting the same issue, which suggests that it is not caused by the script itself.

>it looks like esphome doesn't works on alpine.

Where did you find this info?

**Comment by ntguest** [comment]
actually it tested on ubuntu, debian and armbian ... and in docker
but i cant compile in Alpine LXC

**Comment by tteck** [comment]
As this is a recent addition to the repository, we will hold off and see what others discover before taking any further action.

**Comment by tteck** [comment]
Pulled
